RecoEgamma/EgammaIsolationAlgos/src/EleTkIsolFromCands.cc

```diff
@@ -135,21 +133,20 @@ EleTkIsolFromCands::PIDVeto EleTkIsolFromCands::pidVetoFromStr(const std::string
   }
 }
 
-bool EleTkIsolFromCands::passTrkSel(const reco::TrackBase& trk,
-                                    const double trkPt,
-                                    const TrkCuts& cuts,
-                                    const double eleEta,
-                                    const double elePhi,
-                                    const double eleVZ) {
-  const float dR2 = reco::deltaR2(eleEta, elePhi, trk.eta(), trk.phi());
-  const float dEta = trk.eta() - eleEta;
-  const float dZ = eleVZ - trk.vz();
-
-  return dR2 >= cuts.minDR2 && dR2 <= cuts.maxDR2 && std::abs(dEta) >= cuts.minDEta && std::abs(dZ) < cuts.maxDZ &&
-         trk.hitPattern().numberOfValidHits() >= cuts.minHits &&
+bool EleTkIsolFromCands::passTrackPreselection(const reco::TrackBase& trk, const TrkCuts& cuts) {
+  return trk.hitPattern().numberOfValidHits() >= cuts.minHits &&
          trk.hitPattern().numberOfValidPixelHits() >= cuts.minPixelHits &&
-         (trk.ptError() / trkPt < cuts.maxDPtPt || cuts.maxDPtPt < 0) && passQual(trk, cuts.allowedQualities) &&
-         passAlgo(trk, cuts.algosToReject) && trkPt > cuts.minPt;
+         (trk.ptError() / trk.pt() < cuts.maxDPtPt || cuts.maxDPtPt < 0) && passQual(trk, cuts.allowedQualities) &&
+         passAlgo(trk, cuts.algosToReject) && trk.pt() > cuts.minPt;
```

Comment:
I think ::pt() invokes a sqrt which is called 2 times in this formula.

Reply:
Thanks for spotting this! I'll make sure `::pt()` is only called once.

Reply:
I expect that repeated calls to an object function are optimized away by the compiler, at least for small enough scope of the code and where it's clear that the object is not changing between the calls

Reply:
Right, that might happen. But there is also third call outside the function when the SimpleTracks are filled. So to call `pt()` only once, I'll just do it in `preselectTracks` and then pass the value to `passTrackPreselection` and we don't have to rely on the compiler optimization anyway.

Reply:
and since you are in the process 
```(trk.ptError()  < cuts.maxDPtPt *  trk.pt() ```
and in any case all that code is dominated by this
https://cmssdt.cern.ch/dxr/CMSSW/source/DataFormats/TrackReco/interface/TrackBase.h#697

(and NO the compiler will NOT optimize pt()*pt()* p() * p() / charge() / charge() )
and will recompute again in the line below  pt() * p() / charge()  given that the order is different
(maybe with Ofast, do we like Ofast these days???)

So, if anything optimize the computation of ptError or even better of ptError/pt that is equivalent to deltaP/(1/p) that is what is stored and the only meaningful quantity for tracks